dfastmi/batch/AConfigurationInitializerBase.py

```diff
@@ -0,0 +1,176 @@
+
+    @property
+    def tstag(self) -> float:
+        """Fraction of year during which flow velocity is considered negligible [-]."""
+        return self._tstag
+    
+    @property
+    def ucrit(self) -> float:
+        """Critical flow velocity [m/s]."""
+        return self._ucrit
+    
+    @property
+    def slength(self) -> float:
+        """The expected yearly impacted sedimentation length [m]."""
+        return self._slength
+    
+    @property
+    def time_fractions_of_the_year(self) -> Vector:
+        """A vector of values each representing the fraction of the year during which the discharge is given by the corresponding entry in Q [-]."""
+        return self._time_fractions_of_the_year
+    
+    @property
+    def time_mi(self) -> Vector:
+        """A vector of values each representing the fraction of the year during which the discharge Q results in morphological impact [-]."""
+        return self._time_mi
+    
+    @property
+    def celerity(self) -> Vector:
+        """A vector of values each representing the bed celerity for the period given by the corresponding entry in Q [m/s]."""
+        return self._celerity
+    
+    @property
+    def tide_bc(self) -> Tuple[str, ...]:
+        """Array of tidal boundary condition; one per forcing condition."""
+        return self._tide_bc
+    
+    @property
+    def apply_q(self) -> BoolVector:
+        """A list of flags indicating whether the corresponding entry in Q should be used."""
+        return self._apply_q
+    
+    @property
+    def needs_tide(self) -> bool:
+        """Specifies whether the tidal boundary is needed."""
+        return self._needs_tide
+    
+    @property    
+    def n_fields(self) -> int:
+        """An int stating the number of fields."""
+        return self._n_fields
+    
+    @abstractmethod
+    def init(self,
+        reach: IReach,
+        config: ConfigParser) -> None:
+        """
+        Will initialize the config for this reach
+        """
+
+    def _set_ucrit(self, reach : IReach, config: ConfigParser) -> None:
+        """
+        Set critical flow velocity [m/s] based on dfast mi configuration 
+        and river default configuration or use min default.
+
+        Arguments
+        ---------
+        reach : IReach
+            Reach which we want to do the initialization on.
+        config : ConfigParser        
+            The variable containing the configuration.
+        
+        Return
+        ------
+        None
+        """
+
+        try:
+            ucrit = float(config.get("General", "Ucrit", fallback=""))
+        except ValueError:
+            ucrit = reach.ucritical
+        ucrit_min = 0.01
+        ucrit = max(ucrit_min, ucrit)
+        self._ucrit = ucrit
+    
+    def _set_slenght(self) -> None:
+        """
+        Should only be called AFTER(!) init. 
+        Set the expected yearly impacted sedimentation length 
+        depending on time_mi and celerity.
+
+        Arguments
```

Comment:
do we want to add arguments en return in doc if it is not used?

Reply:
Part of another discussion